runtime/executor/src/main/java/org/apache/nemo/runtime/executor/task/SourceVertexDataFetcher.java

```diff
@@ -104,7 +107,8 @@ private boolean isWatermarkTriggerTime() {
   private Object retrieveElement() throws NoSuchElementException, IOException {
     // Emit watermark
     if (!bounded && isWatermarkTriggerTime()) {
-      return new Watermark(readable.readWatermark());
+      // index=0 as there is only 1 input stream
+      return new WatermarkWithIndex(new Watermark(readable.readWatermark()), 0);
```

Comment:
Why do we emit this here? The index is outgoing edge index between inter task

Reply:
I'll take a look at this.